monai/transforms/intensity/array.py

```diff
@@ -748,7 +752,9 @@ def __call__(self, img: NdarrayOrTensor) -> NdarrayOrTensor:
         img = img * (self.b_max - self.b_min) + self.b_min
         if self.clip:
             img = clip(img, self.b_min, self.b_max)
-        return img
+        ret, *_ = convert_data_type(img, dtype=self.dtype)
```

Comment:
Thanks @Nic-Ma , here should `dtype` be an optional augment?
Then:
```
if self.dtype:
    ret, *_ = convert_data_type(img, dtype=self.dtype)
else:
    ret, *_ = convert_data_type(img, dtype=img.dtype)
```
Some transforms such as `ThresholdIntensity ` does not have this augment, thus if users forget to set the type here, a tensor input may be converted into numpy array.

Reply:
Sure, updated all the dtype of transforms in the latest commit.

Thanks.